### ISSUE TITLE
test_index_download.py test failure

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#613]
On a fresh install:

```bash
$ python -m unittest tests/test_index_download.py
Attempting to initialize pre-built index cacm.
Downloading index at https://github.com/castorini/anserini-data/raw/master/CACM/lucene-index.cacm.tar.gz...
lucene-index.cacm.tar.gz: 2.27MB [00:00, 4.60MB/s]                                                                                                                        
Extracting temp_dir/indexes/lucene-index.cacm.tar.gz into temp_dir/indexes/lucene-index.cacm.e47164fbd18aab72cdc18aecc0744bb1...
Initializing cacm...
.Attempting to initialize pre-built index cacm.
Downloading index at https://github.com/castorini/anserini-data/raw/master/CACM/lucene-index.cacm.tar.gz...
lucene-index.cacm.tar.gz: 2.27MB [00:00, 5.24MB/s]                                                                                                                        
Extracting temp_dir/indexes/lucene-index.cacm.tar.gz into temp_dir/indexes/lucene-index.cacm.e47164fbd18aab72cdc18aecc0744bb1...
Initializing cacm...
F
======================================================================
FAIL: test_default_cache (tests.test_index_download.TestIndexDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jimmylin/workspace/pyserini/tests/test_index_download.py", line 28, in test_default_cache
    self.assertTrue(os.path.exists(os.path.expanduser('~/.cache/pyserini/indexes')))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 2 tests in 1.439s

FAILED (failures=1)
```

I think downloading fails if the `pyserini` directory doesn't already exist:

```
$ ls ~/.cache/
huggingface
```

**Comment by MXueguang** [comment]
this issue should be fixed by https://github.com/castorini/pyserini/pull/616. 
The reason was `os.unsetenv('PYSERINI_CACHE')` is not clearing the env variable, so the index was downloaded to `temp_dir` accidentally (since other test set PYSERINI_CACHE to `temp_dir`).

**Comment by lintool** [comment]
Confirmed. The issue is fixed at `7e0ddef`.
